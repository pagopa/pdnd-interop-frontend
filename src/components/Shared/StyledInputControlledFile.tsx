import React, { ChangeEvent } from 'react'
import { Typography } from '@mui/material'
import { Box, SxProps } from '@mui/system'
import { StyledInputWrapper } from './StyledInputWrapper'
import { FormikSetFieldValue } from '../../../types'
import { useTranslation } from 'react-i18next'

type StyledInputControlledFileProps = {
  name: string
  value: File | null
  error?: string
  setFieldValue: FormikSetFieldValue
  label?: string

  infoLabel?: string | JSX.Element

  sx?: SxProps
}

export function StyledInputControlledFile({
  label,
  infoLabel,

  name,
  value,
  setFieldValue,
  error,

  sx,
}: StyledInputControlledFileProps) {
  const { t } = useTranslation('shared-components', { keyPrefix: 'styledInputControlledFile' })

  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    const target = event.target as HTMLInputElement
    const files = target.files as FileList
    setFieldValue(name, files[0], false)
  }

  return (
    <StyledInputWrapper name={name} error={error} sx={sx} infoLabel={infoLabel}>
      <Box sx={{ display: 'flex', alignItems: 'center' }}>
        <Box sx={{ mr: 3, flexShrink: 0, position: 'relative' }}>
          <input
            name={name}
            type="file"
            id={name}
            style={{ position: 'absolute', width: '100%', height: '100%' }}
            onChange={onChange}
          />

          <Box
            component="label"
            htmlFor={name}
            sx={{
              zIndex: 1,
              cursor: 'pointer',
              display: 'inline-flex',
              alignItems: 'center',
              px: 3,
              py: 1.5,
              position: 'relative',
              borderRadius: 1,
              '&:hover': {
                backgroundColor: 'primary.dark',
              },
            }}
            color="common.white"
            bgcolor="primary.main"
          >
            <Typography
              component="span"
              variant="body2"
              color="common.white"
              sx={{ fontWeight: 700 }}
            >
              {label}
            </Typography>
          </Box>
        </Box>
        <Typography component="span" variant="body2">
          {t('fileSelectLabel')}:{' '}
          <Typography component="span" variant="inherit" sx={{ fontWeight: 700 }} color="text">
            {value ? value.name : t('noFileSelectedLabel')}
          </Typography>
        </Typography>
      </Box>
    </StyledInputWrapper>
  )
}
