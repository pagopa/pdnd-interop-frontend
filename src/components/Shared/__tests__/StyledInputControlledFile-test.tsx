import React from 'react'
import renderer from 'react-test-renderer'
import { useForm } from 'react-hook-form'
import { StyledInputControlledFile } from '../StyledInputControlledFile'

function File() {
  const {
    control,
    formState: { errors },
  } = useForm()

  const props = {
    control,
    errors,
    name: 'file',
    label: 'Test file input',
    rules: {},
  }

  return <StyledInputControlledFile {...props} />
}

describe('Snapshot', () => {
  it('matches', () => {
    const component = renderer.create(<File />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
