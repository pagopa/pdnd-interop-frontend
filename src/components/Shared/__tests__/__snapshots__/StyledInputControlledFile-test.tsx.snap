// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches 1`] = `
<div
  className="MuiBox-root css-62igne"
>
  <div
    className="MuiBox-root css-70qvj9"
  >
    <div
      className="MuiBox-root css-14b8c80"
    >
      <input
        id="file"
        name="file"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          }
        }
        type="file"
      />
      <label
        className="MuiBox-root css-ih0hcs"
        htmlFor="file"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 css-1p88kj8-MuiTypography-root"
        >
          Test file input
        </span>
      </label>
    </div>
    <span
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      File selezionato:
       
      <span
        className="MuiTypography-root MuiTypography-inherit css-19wno5d-MuiTypography-root"
      >
        nessun file selezionato
      </span>
    </span>
  </div>
</div>
`;
